compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SDLFileGenerator.java

```diff
@@ -0,0 +1,391 @@
+    private static final String LIST_FORMAT = "[%s]";
+    private static final String QUERY_FORMAT = "  query: %s";
+    private static final String MUTATION_FORMAT = "  mutation: %s";
+    private static final String SUBSCRIPTION_FORMAT = "  subscription: %s";
+    private static final String DEPRECATE = " @deprecated";
+
+    //Schema delimiters
+    private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+    private static final String INDENTATION = "  ";
+    private static final String EMPTY_STRING = "";
+    private static final String COMMA_SIGN = ", ";
+    private static final String AND_SIGN = " & ";
+    private static final String PIPE_SIGN = "|";
+
+    public SDLFileGenerator(Schema schema, int serviceHashCode, Project project) {
+        this.schema = schema;
+        this.serviceHashCode = serviceHashCode;
+        this.project = project;
+    }
+
+    public void generate() {
+        String schema = getSDLSchemaString();
+        writeFile(schema);
+    }
+
+    private String getSDLSchemaString() {
+        String rootType = createRootType();
+        String directives = getDirectives();
+        String types = getTypes();
+        return getFormattedString(SCHEMA_FORMAT, createDescription(this.schema.getDescription()), rootType, directives,
+                                  types);
+    }
+
+    private String createRootType() {
+        List<String> rootOperations = new ArrayList<>();
+        Type query = this.schema.getQueryType();
+        if (query != null) {
+            rootOperations.add(getFormattedString(QUERY_FORMAT, query.getName()));
+        }
+
+        Type mutation = this.schema.getMutationType();
+        if (mutation != null) {
+            rootOperations.add(getFormattedString(MUTATION_FORMAT, mutation.getName()));
+        }
+
+        Type subscription = this.schema.getSubscriptionType();
+        if (subscription != null) {
+            rootOperations.add(getFormattedString(SUBSCRIPTION_FORMAT, subscription.getName()));
+        }
+        return getFormattedString(ROOT_TYPE_FORMAT, String.join(LINE_SEPARATOR, rootOperations));
+    }
+
+    private String getDirectives() {
+        List<String> directives = new ArrayList<>();
+        for (Directive directive : this.schema.getDirectives()) {
+            if (!isDefaultDirective(directive)) {
+                directives.add(createDirective(directive));
+            }
+        }
+        String formatedDirectives = String.join(LINE_SEPARATOR + LINE_SEPARATOR, directives);
+        if (directives.isEmpty()) {
+            return formatedDirectives;
+        }
+        return LINE_SEPARATOR + LINE_SEPARATOR + formatedDirectives;
+    }
+
+    private String getTypes() {
+        List<String> types = new ArrayList<>();
+        for (Map.Entry<String, Type> entry : this.schema.getTypes().entrySet()) {
+            if (!isIntrospectionType(entry.getValue()) && !isBuiltInScalarType(entry.getValue())) {
```

Comment:
How does this handle custom scalar types? For example, `decimal` type? Shall we add a test for that?